### ISSUE TITLE
Feature suggestion - validate decimal on float

### ISSUE THREAD

**Issue by joubertredrat** [Respect/Validation#744]
Hi guys,

I have a suggestion for FloatVal validate, today, you validate as below:

``` php
v::floatVal()->validate(1.5); // true
v::floatVal()->validate('1e5'); // true
```

But in some cases, we need to validate decimal too, then I proposal to validate decimal opcional as below

``` php
v::floatVal()->validate(1.5); // true
v::floatVal()->validate(1.5)->decimal(1); // true
v::floatVal()->validate(1.5)->decimal(2); // false
v::floatVal()->validate('1e5'); // true
v::floatVal()->validate('1e5')->decimal(5); // false
v::floatVal()->validate('27990.50')->decimal(2); // true
v::floatVal()->validate('27990.50')->decimal(1); // false
```

Then, what you think guys? if like this, talk me, I already forked this repo and I can develop this implementation.

**Comment by asanchezml** [comment]
+1

**Comment by alexconrad** [comment]
+1

**Comment by guihouchang** [comment]
+1

**Comment by henriquemoody** [comment]
Inside the validator, it's not possible to determine the amount of decimals coming from a float if they have ending zeros.

The code below:

```
$input = 2.000;

echo "echo" . PHP_EOL;
echo $input;
echo PHP_EOL . PHP_EOL;

echo "var_export" . PHP_EOL;
var_export($input);
echo PHP_EOL . PHP_EOL;

echo "print_r" . PHP_EOL;
print_r($input);
echo PHP_EOL . PHP_EOL;

echo "var_dump" . PHP_EOL;
var_dump($input);
echo PHP_EOL;

echo "json_encode" . PHP_EOL;
echo json_encode($input);
echo PHP_EOL;
```

Output:

```
echo
2

var_export
2.0

print_r
2

var_dump
float(2)

json_encode
2
```

If we only deal with strings, that wouldn't be a problem, but then the user needs to know too much about the rule. Not sure if it's worth it

**Comment by henriquemoody** [comment]
I created it anyways, you should find it available on version 2.1.


Thanks for the suggestion! 🐼
